extensions/compose/src/cli-run.ts

```diff
@@ -16,177 +16,83 @@
-    const system = process.platform;
-
-    // Before copying the file, make sure it's executable (chmod +x) for Linux and Mac
-    if (system === 'linux' || system === 'darwin') {
-      try {
-        await this.runCommand('chmod', ['+x', binaryPath]);
-        console.log(`Made ${binaryPath} executable`);
-      } catch (error) {
-        throw new Error(`Error making binary executable: ${error}`);
-      }
-    }
-
-    // Create the appropriate destination path (Windows uses AppData/Local, Linux and Mac use /usr/local/bin)
-    // and the appropriate command to move the binary to the destination path
-    let destinationPath: string;
-    let command: string[];
-    if (system === 'win32') {
-      destinationPath = path.join(os.homedir(), 'AppData', 'Local', 'Microsoft', 'WindowsApps', `${binaryName}.exe`);
-      command = ['copy', `"${binaryPath}"`, `"${destinationPath}"`];
-    } else {
-      destinationPath = path.join(localBinDir, binaryName);
-      command = ['cp', binaryPath, destinationPath];
-    }
-
-    // If on macOS or Linux, check to see if the /usr/local/bin directory exists,
-    // if it does not, then add mkdir -p /usr/local/bin to the start of the command when moving the binary.
-    if ((system === 'linux' || system === 'darwin') && !fs.existsSync(localBinDir)) {
-      command.unshift('mkdir', '-p', localBinDir, '&&');
-    }
-
-    // If windows or mac, use sudo-prompt to elevate the privileges
-    // if Linux, use sudo and polkit support
-    if (system === 'win32' || system === 'darwin') {
-      return new Promise<void>((resolve, reject) => {
-        // Convert the command array to a string for sudo prompt
-        // the name is used for the prompt
-        const sudoOptions = {
-          name: 'Binary Installation',
-        };
-        const sudoCommand = command.join(' ');
-        sudo.exec(sudoCommand, sudoOptions, error => {
-          if (error) {
-            console.error(`Failed to install '${binaryName}' binary: ${error}`);
-            reject(error);
-          } else {
-            console.log(`Successfully installed '${binaryName}' binary.`);
-            resolve();
-          }
-        });
-      });
-    } else {
-      try {
-        // Use pkexec in order to elevate the prileges / ask for password for copying to /usr/local/bin
-        await this.runCommand('pkexec', command);
-        console.log(`Successfully installed '${binaryName}' binary.`);
-      } catch (error) {
-        console.error(`Failed to install '${binaryName}' binary: ${error}`);
-        throw error;
-      }
+  } else {
+    try {
+      // Use pkexec in order to elevate the prileges / ask for password for copying to /usr/local/bin
+      await extensionApi.process.exec('pkexec', command);
+      console.log(`Successfully installed '${binaryName}' binary.`);
+    } catch (error) {
+      console.error(`Failed to install '${binaryName}' binary: ${error}`);
+      throw error;
```

Comment:
Remove console.error call just throw an Error the same way it is done for chmod call above to avoid printing error to console twice. Once here and second time where installBinaryToSystem called.
```suggestion
      throw new Error(`Failed to install '${binaryName}' binary: ${error}`);
```

Reply:
Updated, thanks for pointing this!